routes/transaction.go

```diff
@@ -122,6 +122,116 @@ func (fes *APIServer) GetTxn(ww http.ResponseWriter, req *http.Request) {
 	}
 }
 
+const GetTxnsRequestCountMax = 500
+
+type GetTxnsRequest struct {
+	TxnHashHexes []string  `safeForLogging:"true"`
+	TxnStatus    TxnStatus `safeForLogging:"true"` // If unset, defaults to TxnStatusInMempool.
+}
+
+type GetTxnsResponse struct {
+	// Map of TxnHashHex strings -> TxnFound booleans
+	TxnsFound map[string]bool
+}
+
+func (fes *APIServer) GetTxns(ww http.ResponseWriter, req *http.Request) {
+	// Parse JSON request body.
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := GetTxnsRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetTxns: Problem parsing request body: %v", err))
+		return
+	}
+
+	// Validate the TxnHashHexes param.
+	if len(requestData.TxnHashHexes) == 0 {
+		_AddBadRequestError(ww, fmt.Sprintf("GetTxns: TxnHashHexes is empty."))
+		return
+	}
+	if len(requestData.TxnHashHexes) > GetTxnsRequestCountMax {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetTxns: TxnHashHexes length %d is greater than %d.",
+			len(requestData.TxnHashHexes),
+			GetTxnsRequestCountMax,
+		))
+		return
+	}
+
+	// Validate the TxnStatus param.
+	txnStatus := requestData.TxnStatus
+	if txnStatus == "" {
+		txnStatus = TxnStatusInMempool
+	}
+	if txnStatus != TxnStatusInMempool && txnStatus != TxnStatusCommitted {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetTxns: Invalid TxnStatus: %s. Options are {InMempool, Committed}.", txnStatus,
+		))
+		return
+	}
+
+	// Decode the TxnHashHexes.
+	var txnHashes []*lib.BlockHash
+	for _, txnHashHex := range requestData.TxnHashHexes {
+		txnHashBytes, err := hex.DecodeString(txnHashHex)
+		if err != nil || len(txnHashBytes) != lib.HashSizeBytes {
+			_AddBadRequestError(ww, fmt.Sprintf("GetTxns: Error parsing txn hash %s: %v", txnHashHex, err))
+			return
+		}
+		txnHashes = append(txnHashes, lib.NewBlockHash(txnHashBytes))
+	}
```

Comment:
```suggestion
	txnHashes := make(map[string]*lib.BlockHash, len(requestData.TxnHashHexes))
	for _, txnHashHex := range requestData.TxnHashHexes {
		txnHashBytes, err := hex.DecodeString(txnHashHex)
		if err != nil || len(txnHashBytes) != lib.HashSizeBytes {
			_AddBadRequestError(ww, fmt.Sprintf("GetTxns: Error parsing txn hash %s: %v", txnHashHex, err))
			return
		}
		txnHashes[txnHashHex] = lib.NewBlockHash(txnHashBytes)
	}
```
This helps with two things: 
1. If we have duplicates, we only really need to go to the DB onces.
2. the `.String` function on the `BlockHash` type isn't SUPER fast, so we can avoid calling it below since we already have the txn hash hex for it.